mava/systems/parameter_client.py

```diff
@@ -110,12 +111,37 @@ def _adjust_and_request(self) -> None:
         )
         self._copy(self._client.get_parameters(self._get_keys))
 
+    def _async_adjust_and_request(self) -> Optional[futures.Future]:
+        """Set the parameters in the server, then update local params from the server.
+
+        Returns:
+            A future for the parameter get
+        """
+        # parameter server only has `futures` attribute if it is a launchpad node
+        # and it is only a launchpad node if we are running in multiprocess
+        if self._client.store.global_config.multi_process:
+            self._client.futures.set_parameters(  # type: ignore
+                {key: self._parameters[key] for key in self._set_keys},
+            )
+            # not sure how these futures will interact
+            # will it fetch the parameters set above?
+            get_future = self._client.futures.get_parameters(self._get_keys)  # type: ignore # noqa
+            get_future.add_done_callback(lambda ctx: self._copy(ctx.result()))
+
+            return get_future
+        else:
+            self._adjust_and_request()
+            return None
+
     def get_async(self) -> None:
         """Asynchronously updates the parameters with the latest copy from server.
 
         Returns:
             None.
         """
+        # Do not do parallel calls if not multiprocess
+        if not self._client.store.global_config.multi_process:
+            return self.get_and_wait()
```

Comment:
Thoughts on this and the similar ones below? Previously we were using a thread in single process mode, I think this is more correct, but also we can't use a thread in single process mode anymore because the parameter server (`_client`) isn't a launchpad node

Reply:
We can maybe just set the `_async_request=get_and_wait` if not self._client.store.global_config.multi_process? This still allows us to not update at every call as is done with `get_and_wait`. What do you think?

Reply:
It seems that you are already do something like this?

Reply:
As discussed offline added an `DoneFuture` and now this is what is done